### PR TITLE
Document cluster adoption

### PR DESCRIPTION
**Pull request by dgoodwin**
x-ref: https://issues.redhat.com/browse/HIVE-1566

### CODE REVIEW

File: pkg/remoteclient/remoteclient.go [hunk 1]
```diff
@@ -140,7 +140,7 @@ func InitialURL(c client.Client, cd *hivev1.ClusterDeployment) (string, error) {
 // the ClusterDeployment to determine if the remote cluster is reachable.
 func Unreachable(cd *hivev1.ClusterDeployment) (unreachable bool, lastCheck time.Time) {
 	cond := utils.FindClusterDeploymentCondition(cd.Status.Conditions, hivev1.UnreachableCondition)
-	if cond.Status == corev1.ConditionUnknown {
+	if cond == nil || cond.Status == corev1.ConditionUnknown {
```
Comment (2uasimojo) [inline pkg/remoteclient/remoteclient.go:143]: This seems unrelated, should be a separate PR.

File: docs/developing.md [hunk 2]
```diff
@@ -262,24 +261,6 @@ NOTE: The cluster name and domain used to create the certificate must match the
 Example:
 `hiveutil create-cluster mycluster --serving-cert=$HOME/mycluster.crt --serving-cert-key=$HOME/mycluster.key`
```
Comment (2uasimojo) [inline docs/developing.md:263]: Should we keep a link to the new doc from here?

File: docs/cluster-adoption.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+## Adopting ClusterDeployments
+
+It is possible to adopt cluster deployments into Hive. To do so you will need to create a ClusterDeployment with Spec.Installed set to True, no Spec.Provisioning section, and include the following:
+
+  * cluster INFRAID (obtained from `oc get infrastructure cluster -o json | jq .status.infrastructureName`)
+  * cluster ID (obtained from `oc get clusterversion version -o json | jq .spec.clusterID`)
+  * reference to a properly formatted admin kubeconfig Secret:  `oc create secret generic mycluster-admin-kubeconfig --from-file=kubeconfig=/tmp/admin.kubeconfig`
+  * Spec.Platform.YourCloudProvider for your cluster, most importantly region and a properly formatted credentials Secret
+
+Use Spec.PreserveOnDelete if you do not want Hive to deprovision resources when the ClusterDeployment is deleted.
```
Comment (2uasimojo) [inline docs/cluster-adoption.md:12]:
```suggestion
Use `Spec.PreserveOnDelete = true` if you do not want Hive to deprovision resources when the ClusterDeployment is deleted.
```

File: docs/cluster-adoption.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+      credentialsSecretRef:
+        name: my-gcp-creds
+      region: us-east1
+  pullSecretRef:
+    name: pull-secret
+```
+
+## Adopting with hiveutil
+
+hiveutil is a development focused CLI tool which can be build from the hive repo. To adopt a cluster specify the following flags:
```
Comment (2uasimojo) [inline docs/cluster-adoption.md:42]: Link to the hiveutil doc?
```suggestion
[hiveutil](hiveutil.md) is a development focused CLI tool which can be build from the hive repo. To adopt a cluster specify the following flags:
```

File: docs/cluster-adoption.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+```bash
+bin/hiveutil create-cluster --base-domain=example.com mycluster --adopt --adopt-admin-kubeconfig=/path/to/cluster/admin/kubeconfig --adopt-infra-id=[INFRAID] --adopt-cluster-id=[CLUSTERID]
```
Comment (2uasimojo) [inline docs/cluster-adoption.md:45]: Is there an option to set `PreserveOnDelete` via hiveutil?

...looks like not :(  Should there be?

File: docs/cluster-adoption.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
+# Cluster Adoption
```
Comment (2uasimojo) [inline docs/cluster-adoption.md:1]: We should link to this doc from... somewhere. Do we consider adoption a dev-only thing? If so, perhaps the suggested link from developing.md is sufficient. Otherwise maybe from using-hive.md?

Comment (2uasimojo) [reply docs/cluster-adoption.md:1]: > Do we consider adoption a dev-only thing?

Looking at the [card](https://issues.redhat.com/browse/HIVE-1566) I can see the answer is "not anymore", so...

File: pkg/remoteclient/remoteclient.go (hunk 1 above)
Comment (dgoodwin) [reply pkg/remoteclient/remoteclient.go:143]: We actually crash if you adopt without this, I will retitle the PR.

File: docs/cluster-adoption.md (hunk 5 above)
Comment (dgoodwin) [reply docs/cluster-adoption.md:45]: Probably not, we agreed at the outset it was never meant for 1-1 compat with the API, and you could use -o yaml to make edits you need. I still consider this pretty fringe and not really necessary for dev focused hiveutil.

File: docs/developing.md (hunk 2 above)
Comment (dgoodwin) [reply docs/developing.md:263]: I'll add a link in the top level README

File: docs/cluster-adoption.md (hunk 6 above)
Comment (dgoodwin) [reply docs/cluster-adoption.md:1]: I'm thinking top level readme. Thought looking at what's there, maybe using hive is a better fit.

File: docs/developing.md (hunk 2 above)
Comment (dgoodwin) [reply docs/developing.md:263]: Ended up moving the whole thing into using-hive.md given it's size.

**Comment by dgoodwin** [comment]
Updated.

**Comment by 2uasimojo** [comment]
lgtm

Leaving for assignee to stamp.

Retitle/adjust description any time (can be done after merge, so no reason to hold this up). Is there a bug report we can link to?

**Comment by dgoodwin** [comment]
No bug report no, this wasn't really a supported thing until this PR but it sounds like ACM has some use for it, thus why I started testing and found the bug.
